src/components/admin/Settings/StorageMappings/Store/types.ts

```diff
@@ -0,0 +1,24 @@
+import { PostgrestError } from '@supabase/postgrest-js';
+import { JsonFormsData } from 'types';
+
+export interface StorageMappingState {
+    provider: string;
```

Comment:
I feel like this should probably just be typed based on the mapping enum since we know we do not want to support anything that isn't explicitly coded for.

Reply:
Funnily enough, the typing became a cumbersome to implement but I can take another stab at it with fresh eyes.

Reply:
Ah, I remember the friction in changing the type: the `provider` property would need to be defaulted to a member of the  `CloudProviderCodes` enumeration to use it as the sole type annotation. If you are comfortable with `provider` being nullable as well then there is no issue here.